app/controllers/donations_controller.rb

```diff
@@ -0,0 +1,42 @@
+class DonationsController < ApplicationController
+  def new
+    @html_id = 'page'
+    @body_id = 'support'
+    @title   = I18n.t('views.donations.support.heading')
+  end
+
+  def thanks
+    @html_id = 'page'
+    @body_id = 'support'
+    @title   = I18n.t('views.donations.thanks.heading')
+  end
+
+  def create
+    amount = params[:amount].tr('$', '')
+
+    customer = Stripe::Customer.create(
+      email: params['js-stripe-email'],
+      source: params['js-stripe-token']
+    )
+
+    if params[:monthly] == 'true'
+      Stripe::Subscription.create(
+        customer: customer.id,
+        plan:     STRIPE_MONTHLY_PLAN_ID,
+        quantity: amount
+      )
+    else
+      Stripe::Charge.create(
+        customer:    customer.id,
+        amount:      amount.to_i * 100,
+        description: t('views.donations.descriptionl_one_time'),
+        currency:    'usd'
```

Comment:
should this come from the front-end in case someone is more comfortable with a different local currency

Reply:
Good question! I think for now we can just roll with `usd` and maybe have other currencies be a donations v2 feature. Anyone else have an opinion about that?

Reply:
I'm fine with being USD only for now. We'll let people tell us that they want local currency support to figure out if we need to build it.